tests/foreman/api/test_repository.py

```diff
@@ -639,8 +641,25 @@ def test_negative_create_checksum(self):
 
         :CaseImportance: Critical
         """
-        with self.assertRaises(HTTPError):
-            entities.Repository(checksum_type=gen_string('alpha')).create()
+        for downloadpolicy in 'background', 'immediate':
+            with self.assertRaises(HTTPError):
+                entities.Repository(checksum_type=gen_string('alpha'),
+                                    download_policy=downloadpolicy).create()
+
+    @tier1
+    @run_only_on('sat')
+    def test_negative_create_checksum_with_on_demand_policy(self):
+        """Attempt to create repository with checksum and on_demand policy.
+
+        :id: de8b157c-ed62-454b-94eb-22659ce1158e
+
+        :expectedresults: A repository is not created and error is raised.
+
+        :CaseImportance: Critical
+        """
+        for checksum_type in 'sha1', 'sha256':
+            with self.assertRaises(HTTPError):
+                entities.Repository(checksum_type=checksum_type).create()
 
```

Comment:
The same negative test case can also be applied while the repository update? I'm ok if we are using the same endpoint with the same HTTP method.

Reply:
yeah , could add one more test around with already created repo.